File: skill/visualization-2026718-v1/assets/recipes/publication-export-r-v1/recipe.R
export_publication_plot <- function(plot, path, width, height, units = "in", dpi = 600, bg = "white") {
  if (!inherits(plot, c("ggplot", "patchwork"))) stop("plot must be a ggplot or patchwork object.")
  if (!is.character(path) || length(path) != 1L || !nzchar(path)) stop("path must be an explicit non-empty file path.")
  ext <- tolower(tools::file_ext(path))
  if (!ext %in% c("pdf", "svg", "png", "tif", "tiff")) stop("Supported formats: pdf, svg, png, tif, tiff.")
  if (!is.numeric(width) || !is.numeric(height) || width <= 0 || height <= 0) stop("width and height must be positive.")
  ggplot2::ggsave(filename = path, plot = plot, width = width, height = height, units = units, dpi = dpi, bg = bg, limitsize = TRUE)
  invisible(normalizePath(path, winslash = "/", mustWork = FALSE))
}

File: skill/visualization-2026718-v1/assets/recipes/publication-export-python-v1/recipe.py
"""Explicit Matplotlib publication export."""
from pathlib import Path


def export_publication_figure(figure, path, width: float, height: float, dpi: int = 600, transparent: bool = False):
    destination = Path(path)
    if destination.suffix.lower() not in {".pdf", ".svg", ".png", ".tif", ".tiff"}:
        raise ValueError("Supported formats: PDF, SVG, PNG, TIF, TIFF")
    if width <= 0 or height <= 0 or dpi <= 0:
        raise ValueError("width, height and dpi must be positive")
    figure.set_size_inches(width, height, forward=True)
    figure.savefig(destination, dpi=dpi, bbox_inches="tight", transparent=transparent)
    return destination
